src/tactic/linarith.lean

```diff
@@ -271,6 +271,60 @@ get_var_list >>= list.mmap' monad.elim_var
 
 end fm_elim
 
+/-!
+`linarith` computes the linear form of its input expressions,
+assuming (without justification) that the type of these expressions
+is a commutative semiring.
+
+It identifies atoms up to ring-equivalence: that is, `(y*3)*x` will be identified `3*(x*y)`,
+where the monomial `x*y` is the linear atom.
+
+* Variables are represented by natural numbers.
+* Monomials are represented by `monom := rb_map ℕ ℕ`. The monomial `1` is represented by the empty map.
+* Linear combinations of monomials are represented by `sum := rb_map monom ℤ`.
+
+All input expressions are converted to `sum`s, preserving the map from expressions to variables.
+We then discard the monomial information, mapping each distinct monomial to a natural number.
+The resulting `rb_map ℕ ℤ` represents the ring-normalized linear form of the expression.
+-/
+
+/-- Variables (represented by natural numbers) map to their power. -/
+@[reducible] meta def monom := rb_map ℕ ℕ
```

Comment:
Should this include an explicit type? Same for `sum` below. I guess there are also some other declarations with missing types above, but I'll complain about those in the next documentation PR.